std/base64.d

```diff
@@ -405,23 +405,23 @@ template Base64Impl(char Map62th, char Map63th, char Padding = '=')
 
         immutable blocks = srcLen / 3;
         immutable remain = srcLen % 3;
-        auto      srcptr = source.ptr;
-        size_t    pcount;
+        auto s = source; // copy for out contract length check
```

Comment:
Because `source` can be a static array I think this has to be `auto s = source[];`

Reply:
If I'm not wrong about this then there probably needs to be a unittest where the source is a static array, since the tests didn't catch this.